### PR TITLE
Issue #266 & #325 - Changed create operation request schema.

### PR DESCRIPTION
**Pull request by thehenrytsai**
1. Changed create operation request schema so DID construction.
1. Updated all affected tests.
1. Refactored `RequestHandler` tests.

### CODE REVIEW

File: lib/core/versions/latest/models/InternalDocumentModel.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Defines the internal generic document data structure that is agnostic to any particular external schema.
+ */
+export default interface InternalDocumentModel {
+  didUniqueSuffix: string;
```
Comment (isaacJChen) [inline lib/core/versions/latest/models/InternalDocumentModel.ts]: Shouldn't this just be `uniqueID`/ `uniqueSuffix` or something not related to DID now?

File: lib/core/versions/latest/RequestHandler.ts [hunk 2]
```diff
@@ -27,7 +30,7 @@ export default class RequestHandler implements IRequestHandler {
     console.info(`Handling operation request of size ${request.length} bytes...`);
 
     // Perform common validation for any write request and parse it into an `Operation`.
-    let operation: Operation;
+    let operation: Operation | OperationModel;
```
Comment (isaacJChen) [inline lib/core/versions/latest/RequestHandler.ts]: Shouldn't parse always return type `OperationModel` on line `operation = await Operation.parse(request);`, so why does this need to be type Operation | OperationModel?

File: tests/generators/OperationGenerator.ts [hunk 3]
```diff
@@ -284,28 +291,62 @@ export default class OperationGenerator {
     return Buffer.from(JSON.stringify(operationJws));
   }
 
+  /**
+   * Creates a named anchored operation model from `OperationModel`.
+   */
+  public static createNamedAnchoredOperationModelFromOperationModel (
```
Comment (isaacJChen) [inline tests/generators/OperationGenerator.ts]: This functions is not used here. Will it be used in later PR?

File: docs/protocol.md [hunk 4]
```diff
@@ -300,42 +279,44 @@ POST / HTTP/1.1
 | --------------------- | ---------------------- |
 | ```Content-Type```    | ```application/json``` |
 
-#### Request body schema
+
+#### Create operation request schema
```
[the remark this answers is not in the document]
Comment (thehenrytsai) [reply docs/protocol.md]: Yes, will do!

File: lib/core/versions/latest/BatchWriter.ts [hunk 5]
```diff
@@ -24,18 +23,16 @@ export default class BatchWriter implements IBatchWriter {
...
     // Do nothing if there is nothing to batch together.
-    if (operationBuffers.length === 0) {
+    if (queuedOperations.length === 0) {
       return;
     }
 
-    const batch = operationBuffers.map(
-      (buffer) => Operation.create(buffer)
-    );
+    const operationBuffers = queuedOperations.map(operation => operation.operationBuffer);
```
[the remark this answers is not in the document]
Comment (thehenrytsai) [reply lib/core/versions/latest/BatchWriter.ts]: Changed to suggested.

File: lib/core/versions/latest/models/InternalDocumentModel.ts (hunk 1 above)
Comment (thehenrytsai) [reply lib/core/versions/latest/models/InternalDocumentModel.ts]: We are still registering "DIDs", so the unique portion is still technically the "didUniqueSuffix".

File: lib/core/versions/latest/RequestHandler.ts (hunk 2 above)
Comment (thehenrytsai) [reply lib/core/versions/latest/RequestHandler.ts]: You are right, it was at one point in development incompatible, but it is now! Updated.

File: tests/generators/OperationGenerator.ts (hunk 3 above)
Comment (thehenrytsai) [reply tests/generators/OperationGenerator.ts]: It is being used in tests.

**Pull request merged**